fcrepo-kernel-api/src/main/java/org/fcrepo/kernel/api/RdfContext.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2015 DuraSpace, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.fcrepo.kernel.api;
+
+/**
+ * A collection of RDF contexts that can be used to extract triples from FedoraResources
+ *
+ * @author acoburn
+ * @since Dec 4, 2015
+ */
+public enum RdfContext {
```

Comment:
@barmintor in one sense these are two separate tickets: 1) change RdfStream to be Stream-based instead of Iterator-based and 2) decouple the modeshape-based RdfStream implementations from the HTTP layers.

Approaching these separately, however, would mean completely rewriting the RdfStream implementations (all 18 of them), and then rewriting them again. IMO, this is like removing a band-aid: it is better to do it all at once.